trieste/models/gpflow/interface.py

```diff
@@ -144,9 +144,11 @@ def optimize(self, dataset: Dataset) -> None:
         """
         self.optimizer.optimize(self.model, dataset)
 
-    def log(self) -> None:
+    def log(self, dataset: Dataset) -> None:
         """
         Log model-specific information at a given optimization step.
+
+        :param dataset: The data that can be used to log additional data-based model summaries.
```

Comment:
Should probably be optional and default to None (like acquisition function builders)